commit ce546fd2eacdbd8dc15f3d2ffd9a95661d082919
Author: Abhijeet Kolekar <abhijeet.kolekar@intel.com>
Date:   Wed Nov 19 15:32:22 2008 -0800

    iwl3945 : Fix ad-hoc mode for 3945
    
    Patch fixes the ad-hoc mode by
    1) Removing redundant clear_stations_table which prevented generation of
    beacons.
    2) Setting assoc_id to 1. It was never set so preventing tx flow
    in iwl3945_tx_skb.
    
    Signed-off-by: Abhijeet Kolekar <abhijeet.kolekar@intel.com>
    Signed-off-by: Reinette Chatre <reinette.chatre@intel.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/iwlwifi/iwl3945-base.c b/drivers/net/wireless/iwlwifi/iwl3945-base.c
index 9330dfa..69fda64 100644
--- a/drivers/net/wireless/iwlwifi/iwl3945-base.c
+++ b/drivers/net/wireless/iwlwifi/iwl3945-base.c
@@ -6321,10 +6321,7 @@ static void iwl3945_post_associate(struct iwl3945_priv *priv)
 
 	case NL80211_IFTYPE_ADHOC:
 
-		/* clear out the station table */
-		iwl3945_clear_stations_table(priv);
-
-		iwl3945_add_station(priv, iwl3945_broadcast_addr, 0, 0);
+		priv->assoc_id = 1;
 		iwl3945_add_station(priv, priv->bssid, 0, 0);
 		iwl3945_sync_sta(priv, IWL_STA_ID,
 				 (priv->band == IEEE80211_BAND_5GHZ) ?
